infrastructure/src/main/java/org/corfudb/infrastructure/datastore/InMemoryDataStore.java

```diff
@@ -0,0 +1,44 @@
+package org.corfudb.infrastructure.datastore;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import lombok.Builder;
+import org.corfudb.common.util.ClassUtils;
+
+import java.util.Optional;
+
+/**
+ * In memory data store.
```

Comment:
Please add more information -- for example, in what context do we use InMemoryDataStore? What guarantees does it provide when compared to the other one?